query/aggregate/aggregate.go

```diff
@@ -72,17 +74,8 @@ func groupBy(list api.SeriesList, tags []string) []group {
 	return result
 }
 
-type aggregate int
-
-const (
-	sumAggregate aggregate = iota + 1
-	meanAggregate
-	minAggregate
-	maxAggregate
-)
-
-var aggregateMap = map[aggregate]func([]float64) float64{
-	sumAggregate:
+var aggregateMap = map[string]func([]float64) float64{
```

Comment:
`aggregateMap` is now accessed through `GetAggregate` and `NewAggregate` functions; `GetAggregate` returns `func([]float64) float64, bool` for `ok` checks, and `NewAggregate` takes a name and a function.